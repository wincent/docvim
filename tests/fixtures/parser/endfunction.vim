" This test shows how we disambiguate between `:endf` and `:endfo`.
fu! A()
  for l:item in getqflist()
    echo l:item
  endfo
endf

fu! B()
  for l:item in getqflist()
    echo l:item
  endfor
endf

fu! C()
  for l:item in getqflist()
    echo l:item
  endfo
endfuncti

fu! D()
  for l:item in getqflist()
    echo l:item
  endfo
endfunction
