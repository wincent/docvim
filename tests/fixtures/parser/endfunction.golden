Unit
  [ FunctionDeclaration
      { functionBang = True
      , functionName = "A"
      , functionArguments = ArgumentList []
      , functionAttributes = []
      , functionBody =
          [ GenericStatement "forl:itemingetqflist()"
          , GenericStatement "echol:item"
          , GenericStatement "endfo"
          ]
      }
  , FunctionDeclaration
      { functionBang = True
      , functionName = "B"
      , functionArguments = ArgumentList []
      , functionAttributes = []
      , functionBody =
          [ GenericStatement "forl:itemingetqflist()"
          , GenericStatement "echol:item"
          , GenericStatement "endfor"
          ]
      }
  , FunctionDeclaration
      { functionBang = True
      , functionName = "C"
      , functionArguments = ArgumentList []
      , functionAttributes = []
      , functionBody =
          [ GenericStatement "forl:itemingetqflist()"
          , GenericStatement "echol:item"
          , GenericStatement "endfo"
          ]
      }
  , FunctionDeclaration
      { functionBang = True
      , functionName = "D"
      , functionArguments = ArgumentList []
      , functionAttributes = []
      , functionBody =
          [ GenericStatement "forl:itemingetqflist()"
          , GenericStatement "echol:item"
          , GenericStatement "endfo"
          ]
      }
  ]
